oci/auth/gcp/auth.go

```diff
@@ -64,69 +63,57 @@ func WithProxyURL(proxyURL *url.URL) Option {
 	}
 }
 
+// WithTokenSource sets a custom token source for the client.
+func (c *Client) WithTokenSource(ts oauth2.TokenSource) *Client {
```

Comment:
What's the use case for option?

Reply:
This is primarily to enable passing in a stub for testing. In general use I'm not seeing a reason to use anything other than the default resolve for fetching the token source (that keeps the behavior aligned with conventional GCP client behavior that people will expect)

Reply:
Since our tests use the same `package gcp` as the main code, we don't need to expose an exported API for specifying this mock/stub token source, you can directly assign a value to the unexported field `c.tokenSource` in the tests. I think I prefer it that way, so we don't introduce a new API without a concrete use case for it besides tests.

Reply:
I see that we have a very similar function `WithTokenURL()` here, so I assume you just followed the same style :+1: 

But since we're changing this, I think I prefer not introducing this API publicly.

@darkowlzz Thoughts?

Reply:
> I see that we have a very similar function WithTokenURL() here, so I assume you just followed the same style 👍

Pretty much, yeah. I think the argument to not make that API public without a use case makes sense, the issue I'm running into is that if it's not on a public method I can't override the private field directly from tests that aren't in the same package (specifically `oci/auth/login/login_test`). There's a couple options that come to mind:
* I can create a distinct function in `oci/auth/gcp/auth_test` to return a client with a passed-in token source (functionally not very different, but it can have a super clear name like `NewClientWithStubTokenSource`)
* I can implement the token source override as an `Option` like `WithProxyUrl` (again, not terrifically different in functionality, but that maybe makes it less confusing)

I don't see another way around this without refactoring the `login` and `oci` packages, which seems like it would be a lot to add to this PR. If you've got another idea I'd be happy to give it a shot, my go is a tad rusty, so there may be something I'm not thinking of.

Reply:
> the issue I'm running into is that if it's not on a public method I can't override the private field directly from tests that aren't in the same package (specifically oci/auth/login/login_test).

Fair enough!